store/src/java/com/zimbra/cs/mailbox/Mailbox.java

```diff
@@ -5810,7 +5811,8 @@ private Message addMessageInternal(MailboxLock l, OperationContext octxt, Parsed
             int flags, String[] tags, int conversationId, String rcptEmail, Message.DraftInfo dinfo,
             CustomMetadata customData, DeliveryContext dctxt, StagedBlob staged, MessageCallbackContext callbackContext, String dsId)
     throws IOException, ServiceException {
-        assert l.isWriteLockedByCurrentThread();
+		// at this point isn't possible to know the kind of the lock (reader/writer), we know it after call "lock()" method
+        //assert l.isWriteLockedByCurrentThread();
```

Comment:
This is asserting that `lock` has been called on `l`. `l` knows that it is a write lock because we used `lockFactory.writeLock` to create it.

Reply:
doesn't matter if you put lockFactory.writeLock, at this point there isn't lock yet, just after call lock() you are able to check if the lock is a writer